src/Build/BuildCheck/Infrastructure/BuildCheckBuildEventHandler.cs

```diff
@@ -67,18 +66,14 @@ public void HandleBuildEvent(BuildEventArgs e)
 
     private void HandleBuildSubmissionStartedEvent(BuildSubmissionStartedEventArgs eventArgs)
     {
-        if (_projectRestore.TryGetValue(eventArgs.SubmissionId, out bool isRestoring))
+        if (isRestoring)
         {
-            if (isRestoring)
-            {
-                _projectRestore[eventArgs.SubmissionId] = false;
-            }
+            isRestoring = false;
```

Comment:
Are we sure that there could not be two submissions with restore in a row? As far as I know it should be...

Reply:
Let's play it sure - extract the restore indicating property and set the _isRestoring  based on that

Reply:
I agree